checker/src/main/java/org/checkerframework/checker/index/IndexMethodIdentifier.java

```diff
@@ -52,9 +56,30 @@ public IndexMethodIdentifier(AnnotatedTypeFactory factory) {
     mathMinMethods = TreeUtils.getMethods("java.lang.Math", "min", 2, processingEnv);
     mathMaxMethods = TreeUtils.getMethods("java.lang.Math", "max", 2, processingEnv);
 
+    stringIndexOfMethods = new ArrayList<>(4);
+    stringIndexOfMethods.add(
+        TreeUtils.getMethod("java.lang.String", "indexOf", processingEnv, "java.lang.String"));
+    stringIndexOfMethods.add(
+        TreeUtils.getMethod(
+            "java.lang.String", "indexOf", processingEnv, "java.lang.String", "int"));
+    stringIndexOfMethods.add(
+        TreeUtils.getMethod("java.lang.String", "lastIndexOf", processingEnv, "java.lang.String"));
+    stringIndexOfMethods.add(
+        TreeUtils.getMethod(
+            "java.lang.String", "lastIndexOf", processingEnv, "java.lang.String", "int"));
+
     lengthOfValueElement = TreeUtils.getMethod(LengthOf.class, "value", 0, processingEnv);
   }
 
+  /**
+   * Returns true iff the argument is an invocation of String#indexOf or String#lastIndexOf that
+   * takes a string parameter.
+   */
+  public boolean isStringIndexOf(Tree methodTree) {
```

Comment:
I would name this "isIndexOfString" (and likewise for the private variable) to emphasize that these are not just "String.indexOf" methods, but they are "indexOf(String)" methods.